pkg/kubelet/pleg/generic.go

```diff
@@ -425,8 +461,11 @@ func (g *GenericPLEG) updateCache(ctx context.Context, pod *kubecontainer.Pod, p
 		status.IPs = g.getPodIPs(pid, status)
 	}
 
-	g.cache.Set(pod.ID, status, err, timestamp)
-	return err
+	if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) && isEventedPLEGInUse() {
+		timestamp = status.TimeStamp
```

Comment:
nit: I would add a comment here why the timestamp is overriden to use the the one from runtime as compared to `g.clock.Now()` when evented pleg is in use (or maybe link to the KEP where this was explained - https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3386-kubelet-evented-pleg#timestamp-of-the-pod-status)